linkerd/stack/tracing/src/lib.rs

```diff
@@ -148,23 +108,26 @@ where
     type Future = Instrumented<P::Future>;
 
     fn proxy(&self, svc: &mut S, request: Req) -> Self::Future {
-        let _enter = self.span.enter();
+        let span = self.get_span();
+        let _enter = span.enter();
         trace!(?request, "proxy");
-        self.inner.proxy(svc, request).instrument(self.span.clone())
+        self.inner.proxy(svc, request).instrument(span.clone())
```

Comment:
nit/TIOLI: we could avoid the span clone (atomic ref count bump) if we did:
```rust
let span = self.get_span();
let future = span.in_scope(|| {
    trace!(?request, "proxy");
    self.inner.proxy(svc, request)
});
future.instrument(span.clone())
```
or this, which IMO is the nicest (but requires updating to the latest `tracing` release to get the new [`Span::entered` method](https://docs.rs/tracing/0.1.25/tracing/span/struct.Span.html#method.entered)):
```rust
let span = self.get_span().entered();
trace!(?request, "proxy");
self.inner.proxy(svc, request).instrument(span.exit())
```